homeassistant/components/squeezebox/browse_media.py

```diff
@@ -172,13 +175,71 @@ def _build_response_known_app(
     """Build item for app or radio."""
 
     return BrowseItemResponse(
+        id=item.get("id", ""),
+        title=item["title"],
         child_item_type=search_type,
         child_media_class=browse_data.content_type_media_class[search_type],
         can_play=bool(item["isaudio"] and item.get("url")),
         can_expand=item["hasitems"],
     )
 
 
+def _build_response_favorites(item: dict[str, Any]) -> BrowseItemResponse:
+    """Build item for Favorites."""
+    if "album_id" in item:
+        return BrowseItemResponse(
+            id=str(item["album_id"]),
+            title=item["title"],
+            child_item_type=MediaType.ALBUM,
+            child_media_class=CONTENT_TYPE_MEDIA_CLASS[MediaType.ALBUM],
+            can_expand=True,
+            can_play=True,
+        )
+    if item["hasitems"] and not item["isaudio"]:
+        return BrowseItemResponse(
+            id=item.get("id", ""),
+            title=item["title"],
+            child_item_type="Favorites",
+            child_media_class=CONTENT_TYPE_MEDIA_CLASS["Favorites"],
+            can_expand=True,
+            can_play=False,
+        )
+    return BrowseItemResponse(
+        id=item.get("id", ""),
+        title=item["title"],
+        child_item_type="Favorites",
+        child_media_class=CONTENT_TYPE_MEDIA_CLASS[MediaType.TRACK],
+        can_expand=item["hasitems"],
+        can_play=bool(item["isaudio"] and item.get("url")),
+    )
+
+
+def _get_item_thumbnail(
+    item: dict[str, Any],
+    player: Player,
+    entity: MediaPlayerEntity,
+    item_type: str | MediaType | None,
+    search_type: str,
+    internal_request: bool,
+) -> str | None:
+    """Construct path to thumbnail image."""
+    item_thumbnail: str | None = None
+    if artwork_track_id := item.get("artwork_track_id"):
+        if internal_request:
+            item_thumbnail = player.generate_image_url_from_track_id(artwork_track_id)
+        elif item_type is not None:
+            item_thumbnail = entity.get_browse_image_url(
+                item_type, item.get("id", ""), artwork_track_id
+            )
+        else:
+            item_thumbnail = item.get("image_url")
+    elif search_type in ["Apps", "Radios"]:
+        item_thumbnail = player.generate_image_url(item["icon"])
+    else:
```

Comment:
Make this a catch all if it wasn't set by a condition above. Then you can remove the duplicate code for setting it above. This improves the readability of the code: 

```suggestion
    if item_thumbnail is None:
```